plugins/org.obeonetwork.m2doc/src/org/obeonetwork/m2doc/provider/DiagramProvider.java

```diff
@@ -0,0 +1,26 @@
+package org.obeonetwork.m2doc.provider;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link IDiagramProvider} instances are used to provide diagram's image file from any modeling tool or technology that porvides graphical
+ * representations of models.
+ * 
+ * @author Romain Guider
+ */
+public abstract class DiagramProvider implements IProvider {
```

Comment:
This is cosmetic but the class could be called AbstractDiagramProvider to reflect the abstract character of it.